### DISCUSSION TITLE
Can anyone explain me why its delegate call and not a static call ?

### DISCUSSION THREAD

**Discussion by DevSwayam** [Cyfrin/security-and-auditing-full-course-s23#131] [Q&A] [answered]
### Description: 
There are two functions which are being called by one by one and i dont understand why one is static call and another is delegate call. Two functions :
```javascript
    function getCalculatedFee(IERC20 token, uint256 amount) public view returns (uint256 fee) {
        //slither-disable-next-line divide-before-multiply
        uint256 valueOfBorrowedToken = (amount * getPriceInWeth(address(token))) / s_feePrecision;
        //slither-disable-next-line divide-before-multiply
        fee = (valueOfBorrowedToken * s_flashLoanFee) / s_feePrecision;
    }
```
This logs are generated while testing thunder Loan Redeem function which is 
```javascript 
 function testRedeem() public setAllowedToken hasDeposits {
        uint256 amountToBorrow = AMOUNT * 10;
        uint256 calculatedFee = thunderLoan.getCalculatedFee(tokenA, amountToBorrow);
        /*
        vm.startPrank(user);
        tokenA.mint(address(mockFlashLoanReceiver), calculatedFee);
        thunderLoan.flashloan(address(mockFlashLoanReceiver), tokenA, amountToBorrow, "");
        vm.stopPrank();
        */
        vm.startPrank(liquidityProvider);
        thunderLoan.redeem(tokenA,type(uint256).max);
        vm.stopPrank();
    }
```
## Proof of Logs:

![ThunderLoan_sol_—_Thunder-Loan-Audit](https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/99035115/48dbadc5-e777-4450-a27f-9a647cbc186e)

**Comment by azhar0406** [comment, marked as answer]
You’re calling the ERC1967 proxy and to preserve the storage it will use a delegate call in the fallback function and pass it to the implementation contract.

<img width="892" alt="Excalidraw" src="https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/19214616/089b4a9a-2b49-4214-bf59-980bfa8fe15e">

**Reply by PatrickAlphaC** [reply]
Love this!!!
